ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/columnstats/AlterTableDropColumnStatisticsAnalyzer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.ddl.table.misc.columnstats;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.common.TableName;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.ddl.DDLSemanticAnalyzerFactory.DDLType;
+import org.apache.hadoop.hive.ql.ddl.DDLUtils;
+import org.apache.hadoop.hive.ql.ddl.table.AbstractAlterTableAnalyzer;
+import org.apache.hadoop.hive.ql.ddl.table.AlterTableType;
+import org.apache.hadoop.hive.ql.exec.ColumnStatsDropTask;
+import org.apache.hadoop.hive.ql.exec.TaskFactory;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.ASTNode;
+import org.apache.hadoop.hive.ql.parse.HiveParser;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.plan.ColumnStatsDropWork;
+
+/**
+ * Analyzer for drop column statistics commands.
+ */
+@DDLType(types = {HiveParser.TOK_ALTERTABLE_DROPCOLSTATS, HiveParser.TOK_ALTERPARTITION_DROPCOLSTATS})
+public class AlterTableDropColumnStatisticsAnalyzer extends AbstractAlterTableAnalyzer {
+  public AlterTableDropColumnStatisticsAnalyzer(QueryState queryState) throws SemanticException {
+    super(queryState);
+  }
+
+  @Override
+  protected void analyzeCommand(TableName tableName, Map<String, String> partitionSpec, ASTNode command)
+      throws SemanticException {
+    Table table = getTable(tableName);
+    
+    if (DDLUtils.isIcebergTable(table)) {
```

Comment:
Please don't use `isIcebergTable` because in the compiler it is too specific. I don't understand why is this method exists in the first place. It is an architectural mistake.

IIUC we provide the drop stats function for storage types that store the column stats in HMS. So how about excluding the ones which can store them by themselves?
```
if (table.isNonNative() && table.getStorageHandler().canSetColStatistics(table))
```

There is also a configuration which controls where stats are stored in case of Iceberg:
https://github.com/apache/hive/blob/0e10ff30292265f00a7e7fb9c93f310f4349ac56/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L2238-L2239

What should we do when it is set to `metasore`? Maybe we should allow deleting? WYT?

Reply:
For now I think I will replace `isIcebergTable` with the condition that you suggested, since it is too specific in the compiler. Along the same lines, I think using the conf `HIVE_ICEBERG_STATS_SOURCE` might be too specific too.

Maybe we should rename this property to remove `ICEBERG`, and update the values  to `metastore` and `other` too? But this is beyond the scope of this PR.